### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by TianYong-Goh**

### CODE REVIEW

File: frontend/src/components/HomePage.js [hunk 1]
```diff
@@ -97,58 +95,52 @@ function DifficultySelection({ socket }) {
...
+            sx={{ padding: "1rem 8rem 1rem 8rem", margin: "1rem" }}>
+            Hard
+          </Button>
+          <MatchingDialog
+            initSeconds={30}
+            isOpen={isMatchingDialogOpen}
+            handleClose={stopFindingMatch}
+            matchStatus={matchStatus}
+            failedFindingMatch={failedFindingMatch}
+          />
```
Comment (hopinxian) [inline frontend/src/components/HomePage.js:138]: Could we abstract out the styling or perhaps create reusable components (e.g. buttons)? Some lines look repeated and Im wondering if we would reuse the styling in another page. 

May not be necessary if such button styling is specific to this page.

File: frontend/src/components/layout/Navbar.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+export default function Navbar() {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const handleClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+  const open = Boolean(anchorEl);
```
Comment (hopinxian) [inline frontend/src/components/layout/Navbar.js:24]: I am confused about this part of the code. Why is a boolean const related to a mutable anchorEL?

**Review by hopinxian — CHANGES_REQUESTED** [review]
Some comments made.

File: frontend/src/App.js [hunk 3]
```diff
@@ -1,25 +1,33 @@
...
-import { Box } from "@mui/material";
 import { URL_MATCH_SVC } from "./configs";
 import io from "socket.io-client";
 
+import SignupPage from "./components/user/SignupPage";
+import LoginPage from "./components/user/LoginPage";
+import HomePage from "./components/HomePage";
+import RoomPage from "./components/matching/RoomPage";
+import ForgetPassword from "./components/user/ForgetPasswordPage";
+import ResetPassword from "./components/user/ResetPasswordPage";
```
[the remark this answers is not in the document]
Comment (TianYong-Goh) [reply frontend/src/App.js:10]: yup it is. What about change it to room instead so all the thing that related to room will be inside.

File: frontend/src/components/HomePage.js (hunk 1 above)
Comment (TianYong-Goh) [reply frontend/src/components/HomePage.js:138]: Yup, I actually tried to have a css file however some of them not working when it is in css and having some weird result even they are given a same styling so would need a lot more time for that. For this pr the main purpose is to get every component there in every pages.

File: frontend/src/components/layout/Navbar.js (hunk 2 above)
Comment (TianYong-Goh) [reply frontend/src/components/layout/Navbar.js:24]: this is being use in the given example. I would change it to useState to let it make more sense.

File: frontend/src/App.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (hopinxian) [reply frontend/src/App.js:10]: What are the things that you will be expecting to put in room apart from the RoomPage?

File: frontend/src/components/HomePage.js (hunk 1 above)
Comment (hopinxian) [reply frontend/src/components/HomePage.js:138]: Sure, we just have to keep in mind to improve in the future

File: frontend/src/App.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (TianYong-Goh) [reply frontend/src/App.js:10]: if there is any component related to room page but for now I don't expect anymore component other the 2 that we have now.

**Review by hopinxian — APPROVED** [review]

**Pull request merged**